sympy/integrals/integrals.py

```diff
@@ -444,6 +444,12 @@ def doit(self, **hints):
             function = function.doit(**hints)
         if function.is_zero:
             return S.Zero
+        if len(self.limits) == 1 and len(self.limits[0]) == 3:
+            (x, a, b) = self.limits[0]
+            if any(i.is_infinite for i in (a,b)) and \
+                (function.is_positive or function.is_negative):
+                if function.limit(x,[i for i in (a,b) if i.is_infinite][0]).is_infinite:
```

Comment:
is_infinite could mean positive or negative infinity, but if it is negative infinity, the result should be negative infinity.

Reply:
Yeah i am making the changes, here is what i am currently working on. But i think i should check function.is_nonnegative or function.is_nonpositive since if function is 0 at some x but diverges to oo or -oo at oo/-oo then also the result will be oo. I have come up with something like this. But i have to call solve_univariate_inequality twice to check for nonpositive and nonnegative for x belonging to the integrand limits.

```
# check for divergent integrals
if len(self.limits) == 1 and len(self.limits[0]) == 3:
    (x, a, b) = self.limits[0]
    xdom = Interval(min(a,b),max(a,b))
    # check if one limit is oo and function limit at that is oo
    if any(i.is_infinite for i in (a,b)) and \
        function.limit(x,[i for i in (a,b) if i.is_infinite][0]).is_infinite:
        if solve_univariate_inequality(function>=0, x, domain=xdom, relational=False) == xdom:
            return S.Infinity
        elif solve_univariate_inequality(function<=0, x, domain=xdom, relational=False) == xdom:
            return S.NegativeInfinity
```